java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java

```diff
@@ -289,7 +288,42 @@ public HttpResponse execute(HttpRequest req) throws UncheckedIOException {
 
     BodyHandler<byte[]> byteHandler = BodyHandlers.ofByteArray();
     try {
-      return messages.createResponse(client.send(messages.createRequest(req), byteHandler));
+      URI rawUri = messages.getRawUri(req);
+
+      // This is a workaround to https://bugs.openjdk.org/browse/JDK-8304701
+      // Warning: When this workaround is removed and the j.n.h.HttpClient.Redirect.NORMAL is used
+      // the 301 and 302 redirects will downgrade a POST to a GET request. See the javadoc of
```

Comment:
So this means that even if the bug is fixed in Java we need to keep the check to avoid request downgrades?

Reply:
I could not find any info what kind of redirects are used from providers like saucelabs.com.
If they (or others) are using 301 or 302 redirects the code is needed, if not the code can be removed.

Reply:
Sauce uses 302 if the user does not have sufficient sessions available and they request an additional one. Retries are sent until an available session is found or until a timeout.